src/components/markdownEditor/view/markdownEditorView.js

```diff
@@ -290,6 +296,31 @@ const MarkdownEditorView = ({
     }
   };
 
+  const _handleOnAddLinkPress = () => {
+    insertLinkModalRef.current?.showModal({
+      selectedText: text.slice(selection.start, selection.end),
+      selection: selection,
+    });
+    inputRef.current?.blur();
+  };
+  const _handleOnAddLinkSheetClose = () => {
+    inputRef.current?.focus();
+  };
+  const _handleInsertLink = ({ label, url, selection, isImage }) => {
+    if (url) {
+      applyWebLinkFormat({
```

Comment:
I still strongly thing the `applySnippet` will serve better in this case, can we can simply cook the link text within `insertLinkModal` that would passed just the text to be added in body... let me know if you need to discuss this in details.. it would also help you render plain link preview inside insertLinkModal

Reply:
actually the applySnippet doesn't format the url text according to our need, like we need [label](url) this format for labeled urls and ![name][(img link) this format for images. And for that purpose we need applyWebLinkFormat and I had used that to accomplish the desired goal.